server/error-handling.js

```diff
@@ -0,0 +1,27 @@
+exports.psqlErrorHandler = (err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request" });
+  } else {
+    next(err);
+  }
+};
+
+exports.customErrorHandler = (err, req, res, next) => {
+  if (err.msg && err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+};
+
+exports.serverErrorHandler = (err, req, res, next) => {
+  res.status(500).send({ msg: "500: Internal server error!" });
+};
+
+exports.postErrorHandler = (err, req, res, next) => {
```

Comment:
This doesn't need to be it's own error handler, this can be combined in your psql errors as you are also checking for a psql code here. Nothing wrong with some if/else if or a switch case in your psql handler looking for error codes.